src/components/tx-flow/index.tsx

```diff
@@ -41,18 +43,26 @@ export const TxModalProvider = ({ children }: { children: ReactNode }): ReactEle
     }
 
     const ok = confirm('Closing this window will discard your current progress.')
-    if (ok) {
-      handleModalClose()
-    }
+    if (!ok) return
+
+    // Reject if the flow is closed
+    txDispatch(TxEvent.USER_QUIT, undefined)
+
+    handleModalClose()
   }, [shouldWarn, handleModalClose])
 
   const setTxFlow = useCallback(
-    (txFlow: TxModalContextType['txFlow'], onClose?: () => void, shouldWarn?: boolean) => {
-      setFlow(txFlow)
+    (newTxFlow: TxModalContextType['txFlow'], onClose?: () => void, shouldWarn?: boolean) => {
+      // Reject if a flow is open and the user changes to a different one
+      if (txFlow && txFlow !== newTxFlow && newTxFlow?.type !== SuccessScreen) {
+        txDispatch(TxEvent.USER_QUIT, undefined)
+      }
+
+      setFlow(newTxFlow)
```

Comment:
We could use the callback syntax here to avoid depending on txFlow, i.e.
```suggestion
      setFlow((prevFlow) => {
        // Reject if a flow is open and the user changes to a different one
        if (prevFlow && prevFlow !== newTxFlow && newTxFlow?.type !== SuccessScreen) {
          txDispatch(TxEvent.USER_QUIT, undefined)
        }
        return newTxFlow
    })
```

Reply:
Addressed in aec7801d2a6e83ca98b0a4588f6e52927ed59792.